Fix API e2e Flake TestDeleteClusterBeforeIsUp

**User Story**
https://public-prow.loodse.com/view/gs/prow-dev-public-data/pr-logs/pull/kubermatic_kubermatic/5927/pre-kubermatic-api-e2e/1308477593527259136

```
=== RUN   TestDeleteClusterBeforeIsUp
=== RUN   TestDeleteClusterBeforeIsUp/delete_cluster_before_is_up
    runner.go:79: Creating project cdg9r4jbnw...
    client.go:74: POST /api/v1/projects
    client.go:74: GET /api/v1/projects/pbl9mcjmnx
    client.go:74: GET /api/v1/projects/pbl9mcjmnx
    runner.go:97: Created project and it became ready after 1.106689726s
    runner.go:446: Creating DigitalOcean cluster "pghjf8bf8s" (v1.18.8, 0 nodes)...
    client.go:74: POST /api/v1/projects/pbl9mcjmnx/dc/kubermatic/clusters
    runner.go:453: Cluster created successfully.
    client.go:74: GET /api/v1/projects/pbl9mcjmnx/dc/kubermatic/clusters/7gnh4mk9mr/health
    util.go:85: Deleting cluster 7gnh4mk9mr...
    runner.go:460: Deleting cluster 7gnh4mk9mr...
    client.go:74: DELETE /api/v1/projects/pbl9mcjmnx/dc/kubermatic/clusters/7gnh4mk9mr
    util.go:87: Failed to delete cluster: [DELETE /api/v1/projects/{project_id}/dc/{dc}/clusters/{cluster_id}][409] deleteCluster default  &{Error:0xc0004c8960}
    util.go:63: Deleting project pbl9mcjmnx...
    runner.go:172: Deleting project...
    client.go:74: DELETE /api/v1/projects/pbl9mcjmnx
    runner.go:182: Project deleted successfully
    util.go:69: Waiting 3m0s for project to be gone...
    client.go:74: GET /api/v1/projects/pbl9mcjmnx
    client.go:74: GET /api/v1/projects/pbl9mcjmnx
    util.go:79: Project deleted successfully after 1.107381948s
--- FAIL: TestDeleteClusterBeforeIsUp (3.81s)
    --- FAIL: TestDeleteClusterBeforeIsUp/delete_cluster_before_is_up (3.81s)
```

The test runner should ignore 409 errors for the DeleteCluster() call.

--- a/pkg/test/e2e/api/runner.go
+++ b/pkg/test/e2e/api/runner.go
@@ -460,7 +460,7 @@ func (r *runner) DeleteCluster(projectID, dc, clusterID string) error {
 	r.test.Logf("Deleting cluster %s...", clusterID)
 
 	params := &project.DeleteClusterParams{ProjectID: projectID, DC: dc, ClusterID: clusterID}
-	setupParams(r.test, params, 1*time.Second, 3*time.Minute)
+	setupParams(r.test, params, 1*time.Second, 3*time.Minute, http.StatusConflict)
 
 	_, err := r.client.Project.DeleteCluster(params, r.bearerToken)
 	if err != nil {
@@ -1129,7 +1129,7 @@ func (r *runner) UpdateDC(seed, dcToUpdate string, dc *models.Datacenter) (*mode
 		DCToUpdate: dcToUpdate,
 		Seed:       seed,
 	}
-	setupParams(r.test, params, 1*time.Second, 3*time.Minute)
+	setupParams(r.test, params, 1*time.Second, 3*time.Minute, http.StatusBadRequest)
 
 	updatedDC, err := r.client.Datacenter.UpdateDC(params, r.bearerToken)
 	if err != nil {

